# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineIC.cpp
# Commit: 58ea17fc315e
# Full Hash: 58ea17fc315ef8bc3a325a958953a50e047f4499
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-07-30 21:50:30
# Regressor Bug: 1567438
# File Overlap Count: 2
# Description:
#   Bug 1567438 part 1 - Replace the enter-monitor-chain bailout code with type monitoring in C++. r=tcampbell,iain
#   
#   When we resumeAfter a JOF_TYPESET op, we had code to enter the type monitor
#   chain for the bailing op, passing it the top stack value. It's simpler to do
#   this monitoring in C++ in FinishBailoutToBaseline and then treat it as a regular
# ==============================================================================

diff -r cd58aae7d47b -r 58ea17fc315e js/src/jit/BaselineIC.cpp
--- a/js/src/jit/BaselineIC.cpp	Tue Jul 30 15:12:39 2019 +0000
+++ b/js/src/jit/BaselineIC.cpp	Tue Jul 30 12:08:48 2019 +0000
@@ -1091,11 +1091,9 @@
   return typeMonitorFallback->addMonitorStubForValue(cx, frame, types, val);
 }
 
-static MOZ_MUST_USE bool TypeMonitorResult(JSContext* cx,
-                                           ICMonitoredFallbackStub* stub,
-                                           BaselineFrame* frame,
-                                           HandleScript script, jsbytecode* pc,
-                                           HandleValue val) {
+bool TypeMonitorResult(JSContext* cx, ICMonitoredFallbackStub* stub,
+                       BaselineFrame* frame, HandleScript script,
+                       jsbytecode* pc, HandleValue val) {
   AutoSweepJitScript sweep(script);
   StackTypeSet* types = script->jitScript()->bytecodeTypes(sweep, script, pc);
   JitScript::MonitorBytecodeType(cx, script, pc, types, val);